include/deal.II/grid/reference_cell.h

```diff
@@ -793,6 +861,53 @@ ReferenceCell::face_reference_cell(const unsigned int face_no) const
 
 
 
+inline unsigned int
+ReferenceCell::child_cell_on_face(const unsigned int face,
+                                  const unsigned int subface) const
+{
+  AssertIndexRange(face, n_faces());
+
+  if (*this == ReferenceCells::Vertex)
+    {
+      Assert(false, ExcNotImplemented());
+    }
+  else if (*this == ReferenceCells::Line)
+    {
+      Assert(false, ExcNotImplemented());
+    }
+  else if (*this == ReferenceCells::Triangle)
+    {
+      static constexpr unsigned int subcells[3][2] = {{0, 1}, {1, 2}, {2, 0}};
+
+      return subcells[face][subface];
+    }
+  else if (*this == ReferenceCells::Quadrilateral)
+    {
+      Assert(false, ExcNotImplemented());
```

Comment:
Why is this empty? Shouldn't it be possible to call `GeometryInfo`?

Reply:
I just copied together what was in the `internal::ReferenceCell::Base` class hierarchy and this was the only case that was implemented.

In fact, one can have misgivings about the way this function is currently implemented anyway. In `GeometryInfo`, the function takes orientation flags, whereas here it does not. That means that the triangle case above is probably wrong in some cases, and at least that the function will be wrong if it is ever called for quads because it doesn't have the necessary information to call the respective function in `GeometryInfo`. I don't know what to do here, particularly because I don't know whether the orientation flags mean anything for triangles and tets at this point. The function here is only ever called for triangles at the moment.

Reply:
> In GeometryInfo, the function takes orientation flags, whereas here it does not.

This is not good and should be fixed in a follow-up PR! The orientation flags are relevant both for triangles and tets (see all the other functions).

Reply:
OK, I will tackle this as a follow-up then.